spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/core/ListTopicArnResolverCached.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2013-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.core;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+import org.springframework.util.Assert;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.services.sns.SnsClient;
+import software.amazon.awssdk.services.sns.model.Topic;
+
+/**
+ * @author Matej Nedić
+ * @since 3.0.0
+ */
+public class ListTopicArnResolverCached implements TopicArnResolver {
```

Comment:
Cache gets never refreshed which for some use cases can be an issue.

Reply:
Maybe instead of refreshing the cache on misses it could have a TTL and refresh when called after a configurable time.

Reply:
How often arn is changed is questiom than as well. Never except recreating topic. Maybe we should just drop caching.